app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperManager.kt

```diff
@@ -134,6 +142,28 @@ class WallpaperManager(
         }
     }
 
+    fun recordDiscoveredMetric() {
+        val hasSentMetric = settings.wallpapersDiscovered
+        if (!hasSentMetric) {
+            metrics.track(Event.Wallpaper.DiscoveredFeature)
+            settings.wallpapersDiscovered = true
+        }
+    }
+
+    private fun recordWallpaperAppliedMetric(appliedWallpaper: Wallpaper) {
+        metrics.track(Event.Wallpaper.NewWallpaperApplied(appliedWallpaper))
+    }
+
+    private fun recordSelectionMetrics(currentWallpaper: Wallpaper) {
+        val previousWallpaperName = settings.previousWallpaper
+        if (previousWallpaperName != defaultWallpaper.name && currentWallpaper == defaultWallpaper) {
+            metrics.track(Event.Wallpaper.WallpaperResetToDefault)
+            // This metric will continue to be reported unless the previous wallpaper is reset
+            settings.previousWallpaper = currentWallpaper.name
+        }
+        metrics.track(Event.Wallpaper.WallpaperSelected(currentWallpaper))
```

Comment:
Maybe we could move this logic to a separate helper class as it it's just telemetry related and it's making the size of `WallpaperManager` grow that will make a bit harder to maintain in the future.

Reply:
This logic was all removed as part of the data review